src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,60 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.MyValidatorException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_COUNT_OF_CHAR = 6;
+    private static final int MIN_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        checkLogin(user);
+        checkPassword(user);
+        checkAge(user);
+        storageDao.add(user);
+        return user;
+    }
+
+    private void checkAge(User user) {
+        if (user.getAge() == null) {
+            throw new MyValidatorException("Age can not be null");
+        }
+        if (user.getAge() < MIN_AGE) {
+            throw new MyValidatorException("Age cannot be less than " + MIN_AGE
+                    + " ,but you entered: " + user.getAge());
+        }
+    }
+
+    private void checkPassword(User user) {
+        if (user.getPassword() == null) {
+            throw new MyValidatorException("Password can not be null");
+        }
+        if (user.getPassword().isBlank()) {
+            throw new MyValidatorException("The password cannot consist of spaces");
+        }
+        if (user.getPassword().length() < MIN_COUNT_OF_CHAR) {
+            throw new MyValidatorException("Password can't fewer less symbol than "
```

Comment:
```suggestion
            throw new MyValidatorException("Password shouldn't be less than "
```
please fix the same issue in all places